src/library/scala/collection/LinearSeq.scala

```diff
@@ -216,9 +216,9 @@ trait StrictOptimizedLinearSeqOps[+A, +CC[X] <: LinearSeq[X], +C <: LinearSeq[A]
 /** A specialized Iterator for LinearSeqs that is lazy enough for Stream and LazyList. This is accomplished by not
   * evaluating the tail after returning the current head.
   */
-private[collection] final class LinearSeqIterator[A](coll: Seq[A]) extends AbstractIterator[A] {
+private[collection] final class LinearSeqIterator[A](coll: LinearSeqOps[A, LinearSeq, LinearSeq[A]]) extends AbstractIterator[A] {
```

Comment:
is this type change needed? does `LinearSeqOps` not extend `Seq`?

Reply:
No, it does not. I tried to fix the problem by adding `Seq` to `LinearSeqOps` but this caused more problems than it solved.